plugins/com.google.cloud.tools.eclipse.util.test/src/com/google/cloud/tools/eclipse/util/ArtifactRetrieverTest.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package com.google.cloud.tools.eclipse.util;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import org.apache.maven.artifact.versioning.ArtifactVersion;
+import org.apache.maven.artifact.versioning.InvalidVersionSpecificationException;
+import org.apache.maven.artifact.versioning.VersionRange;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ArtifactRetrieverTest {
```

Comment:
Could avoid this if `ArtifactRetriever` had a configurable URL-retrieval (`Function<URL,InputStream>`?).  It would also provide a hook in case we need to do something relating to network proxy (viz our `GoogleApiFactory` and `ProxyFactory` classes).

(Thinking back to our suspicion of some Travis failures being from the XML Schema validation hitting network issues trying to download schemas.)